docs/pairing-architecture.md

```diff
@@ -0,0 +1,283 @@
+## Pairing Flow Architecture
+
```

Comment:
This feels like it needs a short introductory paragraph about what the "pairing flow" is, in order to set the scene for the rest of the document.  It can probably be copy-pasted from one of our existing Google Docs.